src/rpc/ngHandlers/NoRippleCheck.cpp

```diff
@@ -0,0 +1,220 @@
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/protocol/TxFlags.h>
+#include <rpc/RPC.h>
+#include <rpc/ngHandlers/NoRippleCheck.h>
+
+#include <fmt/core.h>
+
+namespace RPCng {
+
+NoRippleCheckHandler::Result
+NoRippleCheckHandler::process(
+    NoRippleCheckHandler::Input input,
+    boost::asio::yield_context& yield) const
+{
+    auto const range = sharedPtrBackend_->fetchLedgerRange();
+    auto const lgrInfoOrStatus = RPC::getLedgerInfoFromHashOrSeq(
+        *sharedPtrBackend_,
+        yield,
+        input.ledgerHash,
+        input.ledgerIndex,
+        range->maxSequence);
+
+    if (auto status = std::get_if<RPC::Status>(&lgrInfoOrStatus))
+        return Error{*status};
+
+    auto const lgrInfo = std::get<ripple::LedgerInfo>(lgrInfoOrStatus);
+    auto const accountID = RPC::accountFromStringStrict(input.account);
+    auto const keylet = ripple::keylet::account(*accountID).key;
+    auto const accountObj =
+        sharedPtrBackend_->fetchLedgerObject(keylet, lgrInfo.seq, yield);
+    if (!accountObj)
+        return Error{RPC::Status{
+            RPC::RippledError::rpcACT_NOT_FOUND, "accountNotFound"}};
+
+    ripple::SerialIter it{accountObj->data(), accountObj->size()};
+    ripple::SLE sle{it, keylet};
+    auto accountSeq = sle.getFieldU32(ripple::sfSequence);
+
+    bool const bDefaultRipple =
+        sle.getFieldU32(ripple::sfFlags) & ripple::lsfDefaultRipple;
+
+    auto const fees = input.transactions
+        ? sharedPtrBackend_->fetchFees(lgrInfo.seq, yield)
+        : std::nullopt;
+
+    auto output = NoRippleCheckHandler::Output();
+    if (input.transactions)
+        output.transactions.emplace(boost::json::array());
+
+    auto const getBaseTx = [&](ripple::AccountID const& accountID,
+                               std::uint32_t accountSeq) {
+        boost::json::object tx;
+        tx[JS(Sequence)] = accountSeq;
+        tx[JS(Account)] = ripple::toBase58(accountID);
+        tx[JS(Fee)] = RPC::toBoostJson(fees->units.jsonClipped());
+        return tx;
+    };
+
+    if (bDefaultRipple & !input.roleGateway)
```

Comment:
Although this yields the same result in this case, I suppose the intention was to use `&&`, not `&` here.